apps/users/views/profile/face.py

```diff
@@ -54,9 +67,19 @@ def form_valid(self, form):
             user = self.get_user_from_session()
             user.face_vector = None
             user.save(update_fields=['face_vector'])
-            auth_logout(self.request)
         except (errors.MFAFailedError, errors.BlockMFAError) as e:
             form.add_error('code', e.msg)
             return super().form_invalid(form)
 
-        return redirect('authentication:login')
+        auth_logout(self.request)
+        return super().form_valid(form)
+
+    def get_success_url(self):
+        message_data = {
+            'title': _('Face recognition disable success'),
+            'message': _('Face recognition disable success, return login page'),
+            'interval': 5,
+            'redirect_url': reverse('authentication:login'),
+        }
+        url = FlashMessageUtil.gen_message_url(message_data)
+        return url
```

Comment:
This is an unoptimized Django view file. Here's some guidance on what could be improved:

1. Import all models and mixins in one location at the top of this file.

2. Use `@property` to avoid repeating `self.__class__.NAME`.

3. Rename `code` to match lowercase variable name.

4. Ensure that `_()` has been called when defining `_lazys`, which should not have any side effects unless you explicitly want them.

5. Add a function docstring for class methods, such as `_validate_code`. This would make the method more meaningful within the class it belongs to.

6. Ensure there are no duplicate imports from third-party packages.

These changes aim to improve readability and maintainability without much impact on functionality or performance. Please verify with your team if these improvements align with existing coding standards prior to incorporating them into production scripts or applications.